pygmt/helpers/utils.py

```diff
@@ -78,6 +80,8 @@ def data_kind(data, x=None, y=None, z=None):
     elif hasattr(data, "__geo_interface__"):
         kind = "geojson"
     elif data is not None:
+        if required_z and data.shape[1] < 3:
```

Comment:
Checking for `data.shape[1]` fails for [xarray.Dataset](https://xarray.pydata.org/en/v0.18.1/generated/xarray.Dataset.html) inputs with `AttributeError: 'Dataset' object has no attribute 'shape'. The `shape` attribute is valid only for `numpy.ndarray`, `pandas.DataFrame`, and maybe a few other PyData objects. Will need to create a bugfix and add some extra tests for `xarray.Dataset` inputs.
`

Reply:
I can submit a PR fix since I think that I suggested this. But I feel that we should not label it as a bug since this code has not yet been included in a release, so it might be confusing to list on the changelog as a bugfix relative to v0.4.1.

Reply:
Ok, I've submitted a bugfix PR at #1523. Have put on the `skip-changelog` label so it won't show up in the changelog :)